dolphin/task_manager/resources.py

```diff
@@ -14,20 +14,36 @@
 
 from oslo_log import log
 
+from dolphin import exception
+from dolphin.drivers import manager as drivermanager
+from dolphin.db.sqlalchemy import api as db
+from dolphin.i18n import _
+
 LOG = log.getLogger(__name__)
 
 
 class StoragePoolTask:
     def __init__(self, context, storage_id):
         self.storage_id = storage_id
         self.context = context
+        self.driver = drivermanager.DriverManager()
 
     def sync(self):
-        LOG.info('Pool sync func...')
-        # TODO:
-        # 1. call the driver.list_pools(context)
-        # 2. update the list_pools info to DB
-        pass
+        """
+        :return:
+        """
+        try:
+            pool = self.driver.list_pools(self.context, self.storage_id)
+            pool_id = pool.get('id')
+            db.pool_update(self.context, pool_id, pool)
```

Comment:
Need to have the logic , if pool alreay present update it, otherwise create new pool entry.